lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -202,6 +203,25 @@ def _check_state():
 
         if _check_state():
             proc = module.run_command([module.get_bin_path('ssh-keygen', True), '-lf', self.path])
+            if not proc[0] == 0:
+                if not os.path.isdir(self.path) and os.access(self.path, os.W_OK):
+                    if not self.force:
+                        module.fail_json(
+                            msg='%s seems not to be a valid key file. If you want this module to overwrite '
+                            'the file at %s please specify the force parameter.' % (self.path, self.path))
+                    else:
+                        return False
+                elif not os.path.isdir(self.path) and not os.access(self.path, os.W_OK):
+                    if not self.force:
+                        module.fail_json(
+                            msg='%s seems not to be write-able. If you want this module to (re)generate the '
+                                'keypair anyways please specify the force parameter.' % (self.path))
+                    else:
+                        os.chmod(self.path, stat.S_IWUSR + stat.S_IRUSR)
```

Comment:
You should only do this if `self.check_mode` is `False`. Also, this should better be done in `self.generate` right before it is regenerated.